File: apps/ios/Sources/Baozi/Models/HomeAnchor.swift
import Foundation

/// Single source of truth for the user-facing `~` on the local codex.
/// Resolves to `/root` inside the iSH Alpine fakefs — what
/// `ishDefaultCwd()` returns.
///
/// Used by `PathDisplay` to shorten `/root/foo` to `~/foo` in the UI, and
/// by the local-server directory picker to scope navigation. Never used
/// for remote-server paths.
///
/// Note: the iOS-side `DirectoryPickerView` walks paths via `FileManager`,
/// which cannot see inside the fakefs. Picker UX for local iSH paths is a
/// separate follow-up.
enum HomeAnchor {
    static let path: String = "/root"
}

File: apps/android/app/src/main/java/com/kris99/baozi/android/state/HomeAnchor.kt
package com.kris99.baozi.android.state

import android.content.Context
import java.io.File

/**
 * Single source of truth for the user-facing `~` on the local Android codex.
 * Resolves to `filesDir/codex-home/workspace` — the same path the Rust
 * in-process bootstrap (`session/connection.rs::prepare_android_in_process_config`)
 * uses as the default local thread cwd.
 *
 * Used by `PathDisplay` to shorten absolute container paths to `~/…` and
 * by the local-server directory picker to scope navigation. Never used for
 * remote-server paths.
 */
object HomeAnchor {
    fun path(context: Context): String {
        val dir = File(File(context.applicationContext.filesDir, "codex-home"), "workspace")
        if (!dir.exists()) {
            dir.mkdirs()
        }
        return dir.absolutePath
    }
}
